### PR TITLE
Add optional filter when samples has few corners

### PR DESCRIPTION
**Pull request by DavoliLuca**
Closes: SW-60742

### CODE REVIEW

File: camera_calibration/nodes/cameracalibrator.py [hunk 1]
```diff
@@ -56,6 +57,17 @@ def optionsValidCharuco(options, parser):
     # TODO: check for fisheye and stereo (not implemented with ChArUco)
     return True
 
+def compute_robustness_treshold(size: List[int], old_robustness_treshold: int = None):
```
Comment (g-gemignani) [inline camera_calibration/nodes/cameracalibrator.py]: If you are adding typing, please specify the return value

**Review by g-gemignani — COMMENTED** [review]
Minor nit from my side

File: camera_calibration/nodes/cameracalibrator.py (hunk 1 above)
Comment (DavoliLuca) [reply camera_calibration/nodes/cameracalibrator.py]: in the end I haven't used the function and therefore I've removed typing and the function itself

File: camera_calibration/src/camera_calibration/calibrator.py [hunk 2]
```diff
@@ -245,23 +246,16 @@ def _get_corners(img, board, refine = True, checkerboard_flags=0):
...
     """
     Get chessboard corners from image of ChArUco board
     """
-    h = img.shape[0]
-    w = img.shape[1]
-
-    if len(img.shape) == 3 and img.shape[2] == 3:
-        mono = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
-    else:
-        mono = img
```
Comment (brinij) [inline camera_calibration/src/camera_calibration/calibrator.py]: you can remove then also the additional empty line here
I guess then cv2.aruco.detectMarkers supports rgb images as well and not just mono as before? Did you check this?

File: camera_calibration/src/camera_calibration/camera_calibrator.py [hunk 3]
```diff
@@ -139,7 +139,7 @@ class CalibrationNode:
     def __init__(self, boards, service_check = True, synchronizer = message_filters.TimeSynchronizer, flags = 0,
                 fisheye_flags = 0, pattern=Patterns.Chessboard, camera_name='', checkerboard_flags = 0,
                 max_chessboard_speed = -1, queue_size = 1, no_gui = False, no_mono_if_stereo = False,
-                mask = False):
+                mask=False):
```
Comment (brinij) [inline camera_calibration/src/camera_calibration/camera_calibrator.py]: I would avoid making these linting changes if they are not done on a whole file... then diff to the upstream will be less

File: camera_calibration/src/camera_calibration/camera_calibrator.py [hunk 4]
```diff
@@ -366,6 +366,7 @@ def on_scale(self, scalevalue):
         if self.c.calibrated:
             self.c.set_alpha(scalevalue / 100.0)
 
+
```
Comment (brinij) [inline camera_calibration/src/camera_calibration/camera_calibrator.py]: same here, just linting

File: camera_calibration/src/camera_calibration/calibrator.py (hunk 2 above)
Comment (DavoliLuca) [reply camera_calibration/src/camera_calibration/calibrator.py]: Documentation doesn't mention it https://docs.opencv.org/3.4/d9/d6a/group__aruco.html#ga061ee5b694d30fa2258dd4f13dc98129:~:text=%E2%97%86-,detectMarkers(),-void%20cv%3A%3Aaruco
I've removed it becaused unused

**Review by carzum — APPROVED** [review]

**Review by g-gemignani — APPROVED** [review]

**Pull request merged**